.github/workflows/ubuntu.yml

```diff
@@ -26,7 +26,8 @@ jobs:
     strategy:
       matrix:
         env_file: [actions-38.yaml, actions-39.yaml, actions-310.yaml, actions-311.yaml]
-        pattern: ["not single_cpu", "single_cpu"]
+        # Prevent the include jobs from overriding other jobs
+        pattern: [""]
```

Comment:
Below you can set

```
PATTERN: ${{ matrix.pattern || '' }}
```

Reply:
Unfortunately, that doesn't work, 
since Github will add the ``matrix.pattern`` from the ``includes`` section to jobs that don't have ``matrix.pattern``, so it has to be set upfront.

Otherwise, you get "missing" jobs.

Reply:
Is that because `matrix.pattern` is still defined in the group concurrency and job name? Because that shouldn't be needed as a group key or job name anymore

Reply:
No, it's because of this.
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations

(Doing cross multiply and counting the include jobs separately get me 4 + 5 = 9 jobs, but because of GHA logic, it comes out to be 6 jobs).

Reply:
Ah I see. I was unaware of the supplanting rules for `include`

This is fine for now; would still be good to remove the unneeded `matrix.pattern` in `name` and `concurrency: group:`

Reply:
Well, the include jobs provide a ``matrix.pattern``, so it's still useful I think (e.g. do differentiate between regular actions-3.8 and the min versions build).